### DISCUSSION TITLE
error signal when running a customised Python runtime

### DISCUSSION THREAD

**Discussion by chiyu1203** [bonsai-rx/bonsai#1585] [Q&A]
Dear Bonsai Discussion Forum
I followed Bonsai - Python Scripting Documentation to run a python script (for frame2TTL, if anyone is interested)
I used the CreateRuntime module and filled in the PythonHome with the path to the virtual environment (python 3.11) and ScriptPath (the python file). All in absolute pathes.
However, when I ran the workflow, I saw an error signal 

> The type initializer for 'Delegates' threw an exception

I looked this up online but I am not sure how to fix this error. Could anyone give me some ideas?
Below is my bonsai environment
Bonsai 2.8.1
Python Scripting Library 0.1.0
Many thanks in advance!
Best,
Chiyu

**Comment by bruno-f-cruz** [comment]
Please include the project files. Thanks

**Reply by chiyu1203** [reply]
Hi Bruno, thanks for your quick response. You mean the bonsai workflow and the python files? I have attached it here.
[frame2bonsai.zip](https://github.com/bonsai-rx/bonsai/files/12868859/frame2bonsai.zip)

**Comment by bruno-f-cruz** [comment]
Seems like you are using a version of python not compatible with the pythonnet backend. Can you try 3.9 or 3.10?

**Reply by chiyu1203** [reply]
Hi Bruno, I think you are probably right. When I used python 3.10, I did not see the same error signal anymore. There was no output or return value from the CreateRuntime either. Therefore I felt strange and decide to open bonsai via command prompt (in the python virutal environment), and ran the workflow. However, this time I saw an error message 
![PXL_20231011_214227496](https://github.com/bonsai-rx/bonsai/assets/16696523/17036792-a5ec-41d2-b604-83d1a78b2855)



Since then, no matter how I opened the workflow (via Bonsai GUI or command prompt), I always bumped into the same error message, so I am confused now...

**Reply by chiyu1203** [reply]
Also may I double-check with you the use of CreateRuntime and Eval module? I felt like I almost understand this new package but somehow I can not see the output I saw when I ran the python script on command prompt (eg. conneting to the frame2TTL device and printout current sensor value). Also because of the Runtime.PythonDLL error, I am not sure if I really set my runtime properly or Eval can only return integer....
Please feel free to have a look on this latest workflow and let me know if you have any thought.
Your advice is greatly appreciated.

[frame2bonsai2.zip](https://github.com/bonsai-rx/bonsai/files/12874926/frame2bonsai2.zip)
